src/components/custom-els/LoadingSpinner/index.ts

```diff
@@ -0,0 +1,62 @@
+import * as styles from './styles.css';
+
+/**
+ * A simple spinner. This custom element has no JS API. Just put it in the document, and it'll
+ * spin. You can configure the following using CSS custom properties:
+ *
+ * --size: Size of the spinner element (it's always square). Default: 28px.
+ * --color: Color of the spinner. Default: #4285f4.
+ * --stroke-width: Width of the stroke of the spinner. Default: 3px.
+ * --delay: Once the spinner enters the DOM, how long until it shows. This prevents the spinner
+ *          appearing on the screen for short operations. Default: 300ms.
+ */
+export default class LoadingSpinner extends HTMLElement {
+  private _delayTimeout: number = 0;
+
+  constructor() {
+    super();
+
+    // Ideally we'd use shadow DOM here, but we're targeting browsers without shadow DOM support.
+    // You can't set attributes/content in a custom element constructor, so I'm waiting a microtask.
+    Promise.resolve().then(() => {
+      this.style.display = 'none';
+      this.innerHTML = '' +
```

Comment:
Think it'd be worth going with something that avoids the HTML parse here?
I'm thinking something sortof like a hyperscript helper:

```js
function h(tag, className, children) {
  const el = document.createElement(tag);
  el.className = className;
  for (const child of children) el.appendChild(child);
  return el;
}

this.appendChild(
  h('div', styles.spinnerContainer, [
    h('div', styles.spinnerLayer, [
      h('div', styles.spinnerCircleClipper + ' ' + styles.spinnerLeft, [
        h('div', styles.spinnerCircle)
      ]),
      h('div', styles.spinnerGapPatch, [
        h('div', styles.spinnerCircle)
      ]),
      h('div', styles.spinnerCircleClipper + ' ' + styles.spinnerRight, [
        h('div', styles.spinnerCircle)
      ])
    ])
  ])
);
```

Reply:
https://jsperf.com/innerhtml-vs-createelement-test/30 - parsing is really fast, so I'm happy to go with readability here.

Reply:
hmm - not sure I actually trust that benchmark. It doesn't seem to hold up:

https://esbench.com/bench/59b3233399634800a0349038
![image](https://user-images.githubusercontent.com/105127/46869457-e4901a00-cdf9-11e8-86eb-f456e65be514.png)

Reply:
agreed on the readability bit though, and I don't think the performance difference ends up being super meaningful either way here given that it only happens on creation.